src/tower.cxx

```diff
@@ -49,19 +49,40 @@
 namespace fastipc {
 namespace {
 
-[[nodiscard]] ClientRequest readClientRequest(std::span<const std::byte>& buf) noexcept {
+[[nodiscard]] constexpr io::expected<std::optional<ClientRequest>>
```

Comment:
Instead of nesting things, can we only use the expected and put `std::errc::invalid_argument` if the contents are garbage?

Reply:
the idea is that if the optional is empty then we need more bytes, if there is an unexpected value then the input is garbage. We can make a new error code that says like "give me more bytes" or something to flatten it